Sources/Orbit/Components/Slider.swift

```diff
@@ -0,0 +1,441 @@
+            .highPriorityGesture(
+                DragGesture(minimumDistance: 1)
+                    .onChanged { value in
+
+                        if isDragging.wrappedValue == false {
+                            dragStartOffset = value.startLocation.x - horizontalPosition
+                            isDragging.wrappedValue = true
+                        }
+
+                        let newHorizontalPosition = value.location.x - dragStartOffset
+
+                        guard newHorizontalPosition < width else {
+                            onValueChange(bounds.upperBound)
+                            return
+                        }
+
+                        guard newHorizontalPosition > 0 else {
+                            onValueChange(bounds.lowerBound)
+                            return
+                        }
+
+                        if buckets.count > 1 {
+                            let currentValue = currentValue(location: newHorizontalPosition, width: width)
+                            guard let bucket = buckets.first(where: { $0.contains(currentValue) }) else { return }
+                            onValueChange(currentValue < bucket.middle ? bucket.lowerBound : bucket.upperBound)
+                        } else {
+                            onValueChange(currentValue(location: newHorizontalPosition, width: width))
+                        }
+                    }
+                    .onEnded { _ in
+                        isDragging.wrappedValue = false
+                        dragStartOffset = 0
+                    }
+            )
+    }
+
+    @ViewBuilder func strokedPath(from: CGFloat, to: CGFloat) -> some View {
+        Path { path in
+            path.move(to: .init(x: from, y: lineCenter))
+            path.addLine(to: .init(x: to, y: lineCenter))
+        }
+        .stroke(Color.blueNormal, lineWidth: lineHeight)
+    }
+
+    private var lineHeight: CGFloat {
+        CGFloat.xxSmall * sizeCategory.controlRatio
+    }
+
+    private var lineCenter: CGFloat {
+        lineHeight / 2
+    }
+
+    private var thumbVerticalOverlapping: CGFloat {
+        (SliderThumb.circleDiameter / 2 - lineCenter) * sizeCategory.controlRatio
+    }
+
+    private var accessibilityHint: String {
+        "\(startLabel) - \(endLabel)"
+    }
+
+    private func updateSelectedRangeIfNeeded() {
+        if let upperBound = secondValue, case .range(let selectedRange) = value {
+            selectedRange?.wrappedValue = min(firstValue, upperBound)...max(firstValue, upperBound)
+        }
+    }
+
+    private func currentValue(location: CGFloat, width: CGFloat) -> Value {
+        bounds.lowerBound + Value(location / width) * bounds.distance
+    }
+
+    private func position(by value: Value, width: CGFloat) -> CGFloat {
+        let position = CGFloat((value - bounds.lowerBound) / bounds.distance) * width
+        return min(width, max(0, position))
+    }
+
+    public init(
```

Comment:
Lets extract inits to `public extension` and put comments to them so we get nice docs. Should be done consistently with other component inits.

Reply:
Updated.

Reply:
I still cannot see the change in gitlab 🤔

Reply:
GitHub, right? It's not yet pushed😅 I will ping you back once all threads will be updated / commented.

Sorry, GitHub doesn't have this fancy GitLab's Start Review feature, so I can notify you only when all threads are updated.